Sources/Private/Experimental/Animations/LayerProperty.swift

```diff
@@ -0,0 +1,170 @@
+    .init(
+      caLayerKeypath: "transform.scale.x",
+      customizableProperty: nil /* currently unsupported */)
+  }
+
+  static var scaleY: LayerProperty<CGFloat> {
+    .init(
+      caLayerKeypath: "transform.scale.y",
+      customizableProperty: nil /* currently unsupported */)
+  }
+
+  static var rotation: LayerProperty<CGFloat> {
+    .init(
+      caLayerKeypath: "transform.rotation",
+      customizableProperty: nil /* currently unsupported */)
+  }
+
+  static var anchorPoint: LayerProperty<CGPoint> {
+    .init(
+      caLayerKeypath: #keyPath(CALayer.anchorPoint),
+      customizableProperty: nil /* currently unsupported */)
+  }
+
+  static var opacity: LayerProperty<CGFloat> {
+    .init(
+      caLayerKeypath: #keyPath(CALayer.opacity),
+      customizableProperty: nil /* currently unsupported */)
+  }
+}
+
+// MARK: CAShapeLayer properties
+
+extension LayerProperty {
+  static var path: LayerProperty<CGPath> {
+    .init(
+      caLayerKeypath: #keyPath(CAShapeLayer.path),
+      customizableProperty: nil /* currently unsupported */)
+  }
+
+  static var fillColor: LayerProperty<CGColor> {
+    .init(
+      caLayerKeypath: #keyPath(CAShapeLayer.fillColor),
+      customizableProperty: .color)
+  }
+
+  static var lineWidth: LayerProperty<CGFloat> {
+    .init(
+      caLayerKeypath: #keyPath(CAShapeLayer.lineWidth),
+      customizableProperty: nil /* currently unsupported */)
+  }
+
+  static var strokeColor: LayerProperty<CGColor> {
+    .init(
+      caLayerKeypath: #keyPath(CAShapeLayer.strokeColor),
+      customizableProperty: .color)
+  }
+
+  static var strokeStart: LayerProperty<CGFloat> {
+    .init(
+      caLayerKeypath: #keyPath(CAShapeLayer.strokeStart),
+      customizableProperty: nil /* currently unsupported */)
+  }
+
+  static var strokeEnd: LayerProperty<CGFloat> {
+    .init(
+      caLayerKeypath: #keyPath(CAShapeLayer.strokeEnd),
+      customizableProperty: nil /* currently unsupported */)
+  }
+}
+
+// MARK: CAGradientLayer properties
+
+extension LayerProperty {
+  static var colors: LayerProperty<[CGColor]> {
+    .init(
+      caLayerKeypath: #keyPath(CAGradientLayer.colors),
+      customizableProperty: nil /* currently unsupported */)
+  }
+
+  static var locations: LayerProperty<[CGFloat]> {
+    .init(
+      caLayerKeypath: #keyPath(CAGradientLayer.locations),
+      customizableProperty: nil /* currently unsupported */)
+  }
+
+  static var startPoint: LayerProperty<CGPoint> {
+    .init(
+      caLayerKeypath: #keyPath(CAGradientLayer.startPoint),
+      customizableProperty: nil /* currently unsupported */)
+  }
+
+  static var endPoint: LayerProperty<CGPoint> {
+    .init(
+      caLayerKeypath: #keyPath(CAGradientLayer.endPoint),
+      customizableProperty: nil /* currently unsupported */)
+  }
+}
+
+// MARK: - CustomizableProperty types
+
+extension CustomizableProperty {
+  static var color: CustomizableProperty<CGColor> {
+    .init(
+      name: ["Color"],
```

Comment:
Interesting idea -- I could see that being useful, although I think this it would require refactoring the existing rendering system (unless you specifically mean sharing the string values like `"Color"`

Reply:
Yeah I just mean the string values to make it clear that they're part of an API contract

Reply:
Good idea, I think that's worth doing. I'll add this as a TODO here for when it's time to add support for the rest of these.